File: problem/Minimum_Absolute_Difference/minAbsDiff.js
/**
 * @param {number[][]} grid
 * @param {number} k
 * @return {number[][]}
 */
var minAbsDiff = function(grid, k) {
    const m = grid.length;
    const n = grid[0].length;
    const ans = Array.from({ length: m - k + 1 }, () => new Array(n - k + 1));

    for (let i = 0; i <= m - k; i++) {
        for (let j = 0; j <= n - k; j++) {
            let v = [];
            for (let x = i; x < i + k; x++) {
                for (let y = j; y < j + k; y++) {
                    v.push(grid[x][y]);
                }
            }

            v.sort((a, b) => a - b);
            v = [...new Set(v)];

            if (v.length <= 1) {
                ans[i][j] = 0;
            } else {
                let mn = Infinity;
                for (let p = 0; p < v.length - 1; p++) {
                    mn = Math.min(mn, v[p + 1] - v[p]);
                }
                ans[i][j] = mn;
            }
        }
    }
    return ans;
};

File: problem/Minimum_Absolute_Difference/minAbsDiff.java
import java.util.*;

class Solution {
    public int[][] minAbsDiff(int[][] grid, int k) {
        int m = grid.length;
        int n = grid[0].length;
        int[][] ans = new int[m - k + 1][n - k + 1];

        for (int i = 0; i <= m - k; i++) {
            for (int j = 0; j <= n - k; j++) {
                List<Integer> v = new ArrayList<>();
                for (int x = i; x < i + k; x++) {
                    for (int y = j; y < j + k; y++) {
                        v.add(grid[x][y]);
                    }
                }

                Collections.sort(v);
                
                List<Integer> uniqueV = new ArrayList<>();
                if (!v.isEmpty()) {
                    uniqueV.add(v.get(0));
                    for (int p = 1; p < v.size(); p++) {
                        if (!v.get(p).equals(v.get(p - 1))) {
                            uniqueV.add(v.get(p));
                        }
                    }
                }

                if (uniqueV.size() <= 1) {
                    ans[i][j] = 0;
                } else {
                    int mn = Integer.MAX_VALUE;
                    for (int p = 0; p < uniqueV.size() - 1; p++) {
                        mn = Math.min(mn, uniqueV.get(p + 1) - uniqueV.get(p));
                    }
                    ans[i][j] = mn;
                }
            }
        }
        return ans;
    }
}